### PR TITLE
Issue 42386: Don't show reserved fields as inferred fields

### PR DESCRIPTION
**Pull request by labkey-susanh**
#### Rationale
When inferring fields from a file for a domain, we separate the response fields into two sets: the reserved fields and the others.  This allows the client to handle these sets differently if desired.

#### Related Pull Requests
* https://github.com/LabKey/labkey-ui-components/pull/463

#### Changes
* Add optional `domainKindName` property to form for `property-inferFields`.
* Update @labkey/components in various packages that use the `DomainForm`

### CODE REVIEW

File: experiment/src/org/labkey/experiment/controllers/property/PropertyController.java [hunk 1]
```diff
@@ -844,6 +849,8 @@ private ApiSimpleResponse getInferDomainResponse(DataLoader loader, Integer numL
             }
 
             response.put("fields", fields);
+            if (!reservedFields.isEmpty())
+                response.put("reservedFields", reservedFields);
```
Comment (cnathe) [inline experiment/src/org/labkey/experiment/controllers/property/PropertyController.java]: probably could just remove the if statement and always send back the reservedFields, even if empty. That way the client code can always expect it.

File: list/src/org/labkey/list/model/ListDomainKind.java [hunk 2]
```diff
@@ -231,7 +232,7 @@ public PropertyStorageSpec getPropertySpec(PropertyDescriptor pd, Domain domain)
     @Override
     public Set<String> getReservedPropertyNames(Domain domain)
     {
-        Set<String> properties = new LinkedHashSet<>();
+        Set<String> properties = new CaseInsensitiveHashSet();
```
Comment (cnathe) [inline list/src/org/labkey/list/model/ListDomainKind.java:235]: Did the other domain kinds already use CaseInsensitiveHashSet?

**Review by cnathe — APPROVED** [review]

File: experiment/src/org/labkey/experiment/controllers/property/PropertyController.java (hunk 1 above)
Comment (labkey-susanh) [reply experiment/src/org/labkey/experiment/controllers/property/PropertyController.java]: OK.  Wasn't sure which I preferred.

File: list/src/org/labkey/list/model/ListDomainKind.java (hunk 2 above)
Comment (labkey-susanh) [reply list/src/org/labkey/list/model/ListDomainKind.java:235]: Yes.

**Pull request merged**